packages/grafana-llm-app/pkg/plugin/openai_provider.go

```diff
@@ -24,6 +24,11 @@ func NewOpenAIProvider(settings OpenAISettings, models *ModelSettings) (LLMProvi
 		Timeout: 2 * time.Minute,
 	}
 	cfg := openai.DefaultConfig(settings.apiKey)
+	// validate url
+	_, err := url.Parse(settings.URL)
+	if err != nil {
+		return nil, fmt.Errorf("invalid url: %w", err)
```

Comment:
Is a hard error the correct thing to do here? I think that means for any plugins with current bad data, or that are provisioned incorrectly then the plugin will fail to load which would also mean the setting cannot be updated? Could we have a soft failure instead?